server/src/main/java/org/apache/druid/client/HttpServerInventoryView.java

```diff
@@ -390,6 +390,10 @@ private void serverAdded(DruidServer server)
     synchronized (servers) {
       DruidServerHolder holder = servers.get(server.getName());
       if (holder == null) {
+        if (!finalPredicate.apply(Pair.of(server.getMetadata(), null))) {
```

Comment:
This makes `rhs` of the predicate argument nullable now, but I believe nullable argument can make things complicated. How about adding a new interface like `registerSegmentCallback(Executor exec, SegmentCallback callback, Predicate<DruidServerMetadata> predicate)` to `FilteredServerInventoryView`?